src/components/profile-settings/UpdateProfileForm.tsx

```diff
@@ -0,0 +1,485 @@
+                        </FormControl>
+                        <FormMessage />
+                      </FormItem>
+                    )}
+                  />
+                </div>
+                <div>
+                  <FormField
+                    control={form.control}
+                    name={'website'}
+                    render={({ field }) => (
+                      <FormItem>
+                        <FormLabel>Website</FormLabel>
+                        <FormControl>
+                          <div className="relative">
+                            <Globe className="absolute left-3 top-3 h-4 w-4 text-muted-foreground" />
+                            <Input
+                              type="text"
+                              className="pl-9 focus-visible:ring-0 focus-visible:ring-transparent focus-visible:ring-offset-0"
+                              placeholder="Enter your website"
+                              {...field}
+                            />
+                          </div>
+                        </FormControl>
+                        <FormMessage />
+                      </FormItem>
+                    )}
+                  />
+                </div>
+              </motion.div>
+
+              <Separator />
+
+              {/* Preferences */}
+              <motion.div
+                variants={containerVariants}
+                initial="hidden"
+                animate="visible"
+                className="space-y-6"
+              >
+                <div className="flex flex-row items-center justify-between rounded-lg border p-4">
+                  <div className="space-y-0.5">
+                    <div className="text-sm font-medium">
+                      <div className="flex items-center gap-2">
+                        <BellOff className="h-4 w-4" />
+                        Stop Communications
+                      </div>
+                    </div>
+                    <div className="text-sm text-muted-foreground">
+                      Disable all marketing communications and newsletters
+                    </div>
+                  </div>
+                  <Switch
+                    checked={form.watch('stopCommunication')}
+                    onCheckedChange={checked =>
+                      form.setValue('stopCommunication', checked, {
+                        shouldDirty: true
+                      })
+                    }
+                  />
+                </div>
+
+                <div className="flex items-center gap-2 rounded-lg border p-4">
+                  <Shield className="h-4 w-4" />
+                  <div className="flex-1">Role</div>
+                  <Badge>User</Badge>
+                </div>
+              </motion.div>
+
+              <div className="flex items-center justify-between">
+                <DeleteAccountDialog />
+                <Button
+                  type="submit"
+                  disabled={isLoading || !isFormDirty}
```

Comment:
you shouldnt be checking if form is dirty, react form handles this for u. checkout eventForm.tsx for more details on how we do it.